### ISSUE TITLE
Restore compatibility with Perl 5.8

### ISSUE THREAD

**Issue by jjn1056** [petdance/carp-assert-more#14]
In the most recent release of this module you
added a small change to the code which requires Perl 5.10+ (the
defined or operator).  Here's the line as on Metacpan:

https://metacpan.org/source/PETDANCE/Carp-Assert-More-1.20/More.pm#L695

I know the Perl community needs to learn to love newer Perl but I
would like to ask if you would reconsider this change and use a
slightly longer version of this line which could eliminate the defined
or op and restore compatibility with Perl 5.8.  I only ask because
Catalyst is very committed to supporting our legacy users, many of
which are at companies where upgrading Perl is not going to happen.
Since the change I'm asking for is trivial and I see nothing else in
the code that is greatly benefiting from Perl 5.10+ I would hope this
is something you could consider.  Our only alternative would be to
scour Catalyst and a number of its more important plugins (such as the
Authentication plugin which nearly everyone uses) to remove
Carp::Assert::More and work around as best we can.  It's already been
a lot of hours of volunteer time compared to what I hope would be just
a few minutes of your effort.

All I think we need to do is change:

(line => https://metacpan.org/source/PETDANCE/Carp-Assert-More-1.20/More.pm#L695)

```
my $desc     = shift // 'Must be a DateTime object';
my $desc = defined($_[0]) ? shift(@_) : 'Must be a DateTime object';
```

change to 

I'll also send a PR against master with the change.  Thanks!

**Comment by petdance** [comment]
You don't need to submit a PR, but if you do, make it against dev.

**Comment by petdance** [comment]
Released in 1.24.
